use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,120 @@
+
+- [SPDK Vhost-user target overview](#spdk-vhost-user-target-overview)
+- [Install and setup SPDK vhost-user target](#install-and-setup-spdk-vhost-user-target)
+- [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices)
+- [Launch a Kata container with SPDK vhost-user-blk device](#launch-a-kata-container-with-spdk-vhost-user-blk-device)
+
+## SPDK Vhost-user Target Overview
+
+The Storage Performance Development Kit (SPDK) provides a set of tools and libraries for writing high performance, scalable, user-mode storage applications.
+
+While, virtio is a efficient way to transport data for virtual environments and guests. It is most commonly used in QEMU VMs, where the QEMU itself exposes a virtual PCI device and the guest OS communicates with it using a specific Virtio PCI driver.
+Vhost is a protocol for devices accessible via inter-process communication. It uses the same virtqueue layout as Virtio to allow vhost devices to be mapped directly to virtio devices. The initial vhost implementation is a part of the Linux kernel and uses ioctl interface to communicate with userspace applications.
+Vhost-user implements the control plane through Unix domain socket to establish virtqueue sharing with a user space process on the same host. So SPDK can expose vhost device via Vhost-user protocol.
+
+SPDK vhost is a Vhost-user slave server. It exposes Unix domain sockets and allows external applications to connect. It is capable of exposing virtualized storage devices to QEMU instances or other arbitrary processes.
+
+Currently, SPDK vhost-user target can exposes these types of virtualized devices:
+
+- vhost-blk
+- vhost-scsi
+- vhost-nvme
+
+For more information, visit [SPDK](https://spdk.io) and [SPDK vhost-user target](https://spdk.io/doc/vhost.htm).
+
+## Install and setup SPDK vhost-user application
+
+### Get source code and build SPDK
+
+Following SPDK [getting started](https://spdk.io/doc/getting_started.html)
+
+### Run SPDK vhost-user target
+
+First, run the SPDK setup.sh script to setup some hugepages for the SPDK vhost target
+application.  This will allocate 4096MiB (4GiB) of hugepages, enough for the SPDK
+vhost target and the virtual machine.
+
+```{.sh}
+HUGEMEM=4096 scripts/setup.sh
+```
+
+Then, take directory "/var/run/kata-containers/vhost-user" as Kata's vhost-user device directory. Make subdirectories for vhost-user sockets and device nodes. For more details, see section [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices).
+
+```{.sh}
+mkdir /var/run/kata-containers/vhost-user/
+mkdir /var/run/kata-containers/vhost-user/block/
+mkdir /var/run/kata-containers/vhost-user/block/sockets/
+mkdir /var/run/kata-containers/vhost-user/block/devices/
+```
+
+Next, start the SPDK vhost target application.  The following command will start vhost
+on CPU cores 0 with all future socket files placed in "/var/run/kata-containers/vhost-user/block/sockets/".
+
+```{.sh}
```

Comment:
I'd not seen this ```` ```{.sh} ```` usage on the backticks before.... we tend to use ```` ```bash ```` or ```` ```sh ```` depending on how we want it to be treated by the CI. Have a look at https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md#code-blocks for more details.

Reply:
Ack. Also add "sudo" for root user cmd.
Thanks for your review, @grahamwhaley